JWT Tokens and Basic UI

Karinon opened the pull request:
Added a basic UI with Login-Feature based on JWT, protected routes and a first dummy API-fetch for /api/{user}/workflows.

I find it relatively hard to "guess" types with Django and have used `Any` twice in one line in `freva/views.py`. I am open for better suggestions...

Karinon committed:
JWT, Website-Login and Header. Still WIP, some issues with the routing

Karinon committed:
JWT, Website-Login and Header. Still WIP, some issues with the routing

Karinon committed:
added a way how to determine if a route exists

Karinon committed:
Merge branch 'frontend-basics' of github.com:FREVA-CLINT/freva2 into frontend-basics

# Conflicts:
#	freva/settings.py
#	freva/urls.py
#	freva/views.py

Karinon committed:
re-added BasicAuthentication for testing-purposes.

Karinon committed:
forgot to delete dummy-endpoint from the urls...

Karinon committed:
fix bug in refresh-token-service

Karinon committed:
added a dummy-page for workflows to show how to use the api later on

Karinon committed:
removed underscores

Karinon committed:
typo

Karinon committed:
it's no longer index2

Karinon committed:
fixed hopefully some issues I caused with poetry.lock

Karinon committed:
sorted imports

br-lewis commented:
It's hard for me to comment on the typescript since I don't really know it but I took a cursory glance through and it seems reasonable enough to me.

As for the type annotations, that looks like the sort of thing that probably can't really be reasonably captured by type annotations since I'm guessing Django allows anyone to put anything into `context`? If so I guess that's fine, at least until the types become more clear later.

Karinon committed:
- added an additional VSCode-setting to the devcontainer which fixes an issue
in vscode not finding the Typescript-Types
- fixed the 'high severity'-issue of one lib by updating it
